test/fetch/test_fetch_waitable.c

```diff
@@ -8,27 +8,31 @@
 #include <emscripten/fetch.h>
 #include <emscripten/emscripten.h>
 
-int main()
-{
+int main() {
   emscripten_fetch_attr_t attr;
   emscripten_fetch_attr_init(&attr);
   strcpy(attr.requestMethod, "GET");
   attr.attributes = EMSCRIPTEN_FETCH_LOAD_TO_MEMORY | EMSCRIPTEN_FETCH_WAITABLE;
-  emscripten_fetch_t *fetch = emscripten_fetch(&attr, "file.dat"); // Starts as asynchronous.
+  emscripten_fetch_t *fetch = emscripten_fetch(&attr, "myfile.dat"); // Starts as asynchronous.
 
   EMSCRIPTEN_RESULT ret = EMSCRIPTEN_RESULT_TIMED_OUT;
-  while(ret == EMSCRIPTEN_RESULT_TIMED_OUT)
-  {
-    /* possibly do some other work; */
-    ret = emscripten_fetch_wait(fetch, 0/*milliseconds to wait. 0 to just poll, INFINITY=wait until completion*/);
+  while (ret == EMSCRIPTEN_RESULT_TIMED_OUT) {
+    // possibly do some other work
+    // milliseconds to wait. 0 to just poll, INFINITY=wait until completion
+    ret = emscripten_fetch_wait(fetch, 1);
+  }
+  if (ret != EMSCRIPTEN_RESULT_SUCCESS) {
```

Comment:
This looks like an additional check, not a cleanup, or am I misreading it?

Reply:
Yeah, I was testing to see if `EMSCRIPTEN_FETCH_WAITABLE` worked at all using this example.  Turns out it doesn't work and hasn't worked since we removed the fetch worker.  But this test was lacking this check and was silently succeeding

Reply:
Or at least it was reporting `status` == 0 rather than the actual error code here.